src/common/libflux/attr.h

```diff
@@ -72,6 +72,20 @@ const char *flux_get_hostbyrank (flux_t *h, uint32_t rank);
  */
 int flux_get_rankbyhost (flux_t *h, const char *host);
 
+typedef struct {
+    char text[128];
+} flux_lookup_error_t;
+
+/* Return a list of hosts/ranks in hostlist/idset form given 'targets'
+ *  in idset/hostlist form. Caller must free returned value.
+ *
+ * Returns NULL on failure with error message in errp->text if errp
+ *   is non-NULL.
+ */
+char *flux_hostmap_lookup (flux_t *h,
+                           const char *targets,
+                           flux_lookup_error_t *errp);
```

Comment:
It may be good to also point out
- the source of the mapping is the broker rank-ordered broker `hostlist` attribute
- idset is a set (unordered, no duplicates), while hostlist is a list (ordered, may contain duplicates)
- if there are multiple ranks per host, this function only maps a host to the first rank on it